warn/warn_docstring.go

```diff
@@ -136,18 +136,31 @@ func parseFunctionDocstring(doc *build.StringExpr) docstringInfo {
 		switch block.lines[0] {
 		case "Args:", "Arguments:":
 			if block.lines[0] == "Arguments:" {
-				info.argumentsLine = block.startLineNo
+				// 'Args:' is preferred over 'Arguments:'
+				info.argumentsPos = build.Position{
+					Line:     block.startLineNo,
+					LineRune: indent,
+				}
 			}
 
-			argIndentation := -1
-			for _, line := range block.lines[1:] {
+			argIndentation := 1000000 // Indentation at which previous arg documentation started
+			for i, line := range block.lines[1:] {
+				// Iterate over line and parse arguments. If the current indentation level is the same as
+				// the indentation level of the previous argument (or lower), assume that the new argument
+				// is being described on this line, otherwise it's a continued description of the previous
+				// argument.
 				newIndentation := countLeadingSpaces(line)
-				if newIndentation <= argIndentation || argIndentation == -1 {
+				if newIndentation <= argIndentation {
+					// Extract the header, e.g. "  my_arg (optional, deprecated):"
 					header := argHeaderRegex.FindString(line)
 					if len(header) > 0 {
-						arg := argRegex.FindString(header)
-						info.args[arg] = true
 						argIndentation = newIndentation
+						// Extract the arg name
+						arg := argRegex.FindString(header)
```

Comment:
argHeaderRegex has groups (parenthesis), but you ignore them and you run a second regexp.
I think `FindStringSubmatch` on argHeaderRegex will give you the argument name.

Reply:
Done.